flink-clients/src/main/java/org/apache/flink/client/FlinkYarnSessionCli.java

```diff
@@ -91,6 +93,7 @@
 
 	public FlinkYarnSessionCli(String shortPrefix, String longPrefix) {
 		QUERY = new Option(shortPrefix + "q", longPrefix + "query", false, "Display available YARN resources (memory, cores)");
+		APPLICATION_ID = new Option(shortPrefix + "id", longPrefix + "applicationId", true, "Attach to running YARN session");
```

Comment:
The `APPLICATION_ID` option is integrated here but the `FlinkYarnSessionCli` doesn't support attaching to existing applications. I think it could be useful to check during instantiation of the YarnSessionCli in the `run(..)` method whether we want to start a new application or resume.